src/main/java/mate/academy/service/UserService.java

```diff
@@ -15,6 +15,12 @@
      * Return <code>null</code> if there is no suitable user
      */
     public User findByEmail(String email) {
-        return null;
+        User result = null;
+        for (int i = 0; i < users.length; i++) {
+            if (users[i].getEmail() == email) {
+                result = users[i];
+            }
+        }
+        return result;
```

Comment:
лучше возвращай юзера как только его нашла, иначе, если таких юзеров будет миллион, а с нужным email будет первый, то ты сделаешь много лишних итераций:

```suggestion
        User userByEmail = null;
        for (User user : users) {
            if (user.getEmail().equals(email)) {
                userByEmail = user;
            }
        }
        return userByEmail;
        for (User user : users) {
            if (user.getEmail().equals(email)) {
                return = user;
            }
        }
        return null;
```